backends/tc/ebpfCodeGen.cpp

```diff
@@ -408,13 +408,171 @@ void TCIngressPipelinePNA::emitLocalVariables(EBPF::CodeBuilder *builder) {
 }
 
 // =====================EBPFPnaParser=============================
+EBPFPnaParser::EBPFPnaParser(const EBPF::EBPFProgram *program, const IR::ParserBlock *block,
+                             const P4::TypeMap *typeMap)
+    : EBPFPsaParser(program, block, typeMap) {
+    visitor = new PnaStateTranslationVisitor(program->refMap, program->typeMap, this);
+}
+
 void EBPFPnaParser::emit(EBPF::CodeBuilder *builder) {
     builder->emitIndent();
     builder->blockStart();
     EBPF::EBPFParser::emit(builder);
     builder->blockEnd(true);
 }
 
+void PnaStateTranslationVisitor::compileExtractField(const IR::Expression *expr,
+                                                     const IR::StructField *field,
+                                                     unsigned alignment, EBPF::EBPFType *type) {
+    unsigned widthToExtract = dynamic_cast<EBPF::IHasWidth *>(type)->widthInBits();
```

Comment:
Is it possible to get nullptr as result after dynamic_cast? Checking before access the result would be safe. Or assert it.

Reply:
addressed